backend/src/main/java/no/hvl/dat251/recipeapp/service/RecipeService.java

```diff
@@ -55,23 +68,42 @@ public Recipe getRecipe(Integer id) {
                     return (int) ((100 * sum2 / (ratings2.size() + 1)) - (100 * sum1 / (ratings1.size() + 1)));
                 })
                 .limit(10)
+                .map(r -> computeRecipeRatings(r, user))
                 .collect(Collectors.toList());
     }
 
     public List<Recipe> searchRecipes(String search) {
-        return recipeRepository.findByNameContainingIgnoreCase(search);
+        User user = userService.getCurrentUser();
+        return recipeRepository.findByNameContainingIgnoreCase(search)
+                .stream().map(r -> computeRecipeRatings(r, user)).collect(Collectors.toList());
     }
 
-    public List<Recipe> searchRecipes(String search, User user) {
-        return recipeRepository.findByNameContainingIgnoreCaseAndUserEquals(search, user);
+    public List<Recipe> searchRecipesByCurrentUser(String search) {
+        User user = userService.getCurrentUser();
+        return recipeRepository.findByNameContainingIgnoreCaseAndUserEquals(search, user)
+                .stream().map(r -> computeRecipeRatings(r, user)).collect(Collectors.toList());
     }
 
     public Comment saveComment(Comment comment) {
+        User user = userService.getCurrentUser();
+        comment.setUser(user);
+        comment.setCreated(Instant.now());
         return commentRepository.save(comment);
     }
 
     public Rating saveRating(Rating rating) {
+        User user = userService.getCurrentUser();
+        rating.setUser(user);
         return ratingRepository.save(rating);
     }
 
+    private Recipe computeRecipeRatings(Recipe recipe, User user) {
+        List<Rating> ratings = recipe.getRatings();
+        if(!ratings.isEmpty()) {
+            recipe.setAverageRating((double) ratings.stream().mapToInt(Rating::getRating).reduce(0, Integer::sum) / ratings.size());
+        }
+        ratings.stream().filter(r -> r.getUser().equalsById(user)).findFirst().ifPresent(r -> recipe.setCurrentUserRating(r.getRating()));
+        return recipe;
+    }
+
```

Comment:
I would maybe split the collection operations into multiple lines. For me it's easier to see what's going on.

```suggestion
    private Recipe computeRecipeRatings(Recipe recipe, User user) {
        List<Rating> ratings = recipe.getRatings();
        if(!ratings.isEmpty()) {
            double averageRating = ratings.stream()
                .mapToInt(Rating::getRating)
                .reduce(0, Integer::sum) / ratings.size()
        
            recipe.setAverageRating(averageRating);
        }
        ratings.stream()
            .filter(r -> r.getUser().equalsById(user))
            .findFirst()
            .ifPresent(r -> recipe.setCurrentUserRating(r.getRating()));
            
        return recipe;
    }
```